### PR TITLE
[#11571] Selective Deadline Extensions - Add extension-related tooltips for students 

### LINKED ISSUE #11571: Selective Deadline Extensions
This issue keeps track of PRs relating to the selective deadline extensions feature. More info can be found on the wiki at https://github.com/TEAMMATES/teammates/wiki/Selective-Deadline-Extensions

We'll be working on a feature branch called [`selective-deadline-extension`](https://github.com/TEAMMATES/teammates/tree/selective-deadline-extension), and this issue will be resolved when the branch is merged into master. More details on maintaining a feature branch can be found in the [maintainer guide](https://github.com/TEAMMATES/teammates-ops/blob/master/maintainer-guide.md#using-a-long-lived-feature-branch).

### PR DESCRIPTION
**Pull request by FergusMok**
Fixes #11571 

The current behavior for students' view 
For submission status, only `Pending` and `Submitted` will have the additional (with Extension). (Closed and Awaiting will not). There's also an additional tooltip for students to check on the original end date. 

Tooltip for date
image

Tooltip for status
image
image

Additionally, on the submission page, there will be an additional (with Extension). 
image

### CODE REVIEW

**Comment by samuelfangjw** [comment]
We might need to wait for @jayasting98's PR. Once that is done, `isOpen` should be true if deadline extension exists for a student. I believe that might change the implementation of this PR.

File: src/web/app/pages-session/session-submission-page/session-submission-page.component.ts [hunk 1]
```diff
@@ -170,8 +173,8 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
               } else {
                 // Valid, unused registration key; load information based on the key
                 this.loadCourseInfo();
-                this.loadPersonName();
-                this.loadFeedbackSession();
+                of(this.loadPersonName())
+                  .subscribe(() => this.loadFeedbackSession());
```
[the remark this answers is not in the document]
Comment (FergusMok) [reply src/web/app/pages-session/session-submission-page/session-submission-page.component.ts:177]: The feedback sessions now requires the person's email to see whether they have extensions or not

**Comment by moziliar** [comment]
There are some conflicts that block the CI pipeline.

File: src/web/app/pages-session/session-submission-page/session-submission-page.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (FergusMok) [reply src/web/app/pages-session/session-submission-page/session-submission-page.component.ts]: Removed this part, as its no longer relevant

**Comment by moziliar** [comment]
It seems that component test has failed

**Review by moziliar — APPROVED** [review]
LGTM

**Comment by damithc** [comment]
@FergusMok I'm OK with the UX. On a related note, isn't our standard practice to indicate tool-tipped text with a dotted underline line?
e.g.,
image

**Comment by FergusMok** [comment]
@damithc Thank you for reviewing, yes, I will make the changes for the underline

File: src/web/services/deadline-extension-helper.ts [hunk 2]
```diff
@@ -97,4 +97,17 @@ export class DeadlineExtensionHelper {
     });
     return record;
   }
+
+  public static getUserFeedbackSessionEndingTimestamp(feedbackSession: FeedbackSession): number {
+    if (DeadlineExtensionHelper.hasUserOngoingExtension(feedbackSession)) {
+      return feedbackSession.submissionEndWithExtensionTimestamp!;
+    }
+    return feedbackSession.submissionEndTimestamp;
+  }
```
Comment (jayasting98) [inline src/web/services/deadline-extension-helper.ts:106]: Just wondering something. Does this mean that once the extension is over, the displayed end time will revert back to the usual unextended end time?

**Comment by jayasting98** [comment]
I am also wondering, are there not gonna be more changes for instructors? Since the submission status is not adjusted for instructors within the back-end in `GetFeedbackSessionsAction`, I thought it would be accounted for in the front-end. Is it no longer necessary?

**Comment by samuelfangjw** [comment]
> I am also wondering, are there not gonna be more changes for instructors? Since the submission status is not adjusted for instructors within the back-end in `GetFeedbackSessionsAction`, I thought it would be accounted for in the front-end. Is it no longer necessary?

Not in scope of this PR, am planning to work on this in the near future. Not as straightforward for instructors, need to come up with another solution because in general instructors who are managing the course need to see the original end time and session status. We do not consider this a major blocker at the moment because instructors can still submit (through the link sent via email).

File: src/web/services/deadline-extension-helper.ts (hunk 2 above)
Comment (samuelfangjw) [reply src/web/services/deadline-extension-helper.ts:106]: @FergusMok just checking, has this been resolved?

Comment (FergusMok) [reply src/web/services/deadline-extension-helper.ts:106]: Thanks for the catch, I've updated for this case and in the screenshots too

**Review by jayasting98 — APPROVED** [review]
Seems good, thanks.

**Review by samuelfangjw — APPROVED** [review]

**Pull request merged**